JBPM-6689: Forms & Stunner - Generate Process & Task Forms

romartin opened the pull request:
Hey @pefernan @hasys 

Here is the forms generation stuff.  See [JBPM-6689](https://issues.jboss.org/browse/JBPM-6689)

This is a work done in collaboration with @pefernan , it provides:
* The BPMN forms generation service integration for Stunner
* A new icon on the toolbox for user tasks in order to generate the forms
* A new drop-down menu item in the editor's toolbar in order to generate forms for the process, selected elements or all forms

Thanks!

romartin commented:
Jenkins please retest this

romartin commented:
Jenkins execute full downstream build

domhanak commented:
@romartin Hello, I tried it in Firefox 52ESR and the icon isn't shown correctly. See picture.
![genarate-tforms-firefox52esr](https://user-images.githubusercontent.com/5937251/36722896-17c511da-1baf-11e8-966e-fb3cfe244b7c.png)

romartin commented:
oops good catch @domhanak , lemme check it, my fault, I use to try FF and this time forgot it:: good point thanks! :+1:

romartin commented:
Hey @domhanak , pushed a new commit that fix the svg rendering issue in FF. Tbh not sure yet if that's a bug on how lienzo renders the images on the canvas, or that's a bug or limitation on FF, but if the `width` and `height` attributes are missing inthe svg declaration, FF does not renders the icon at all. Anyway just added the attributes for now. Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins please retest this

jomarko commented on kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-backend/src/test/resources/META-INF/services/org.uberfire.java.nio.file.spi.FileSystemProvider:
```diff
@@ -0,0 +1,3 @@
+org.uberfire.java.nio.fs.file.SimpleFileSystemProvider  # file system provider, also default (1st)
+org.uberfire.java.nio.fs.jgit.JGitFileSystemProvider
+# org.kie.commons.java.nio.fs.eclipse.EclipseFileSystemProvider  # eclipse provider
```
Just interested why we need this commented option here, for easier debugging in future or something? Just asking , if it is not intentional we could remove.

pefernan replied:
@jomarko well seen looks like a copy&paste... my fault!

jsoltes commented:
@romartin can you please add me here as a reviewer?

hasys commented:
Hi @romartin,

I sent a PR https://github.com/romartin/kie-wb-common/pull/5 to your branch with some notes.

romartin commented:
Hey @jomarko @pefernan @hasys 
Fixed the unnecessary eclipse configuration line.
Please @pefernan sync with @hasys on https://github.com/romartin/kie-wb-common/pull/5, once you agree on the changes I'll provide the commits from the PR here, is it ok?

jomarko commented:
@romartin thank you, my approval still valid.

romartin commented:
Hey @hasys @jsoltes @pefernan 

Just given permissions to @pefernan to work in my repository, so he's now able to manage this PR and add new commits when necessary.

@pefernan please, can you move forward this? Most of the issues commented are related to the backend forms implementations, so not easy for me to check/fix... sorry. Also, tried to build the PR from @hasys on top of this (https://github.com/romartin/kie-wb-common/pull/5) and it doesn't compile, it complains about  
 some `FormLayoutGenerator` interace, so please can you cherry-pick his commit as well and merge it on top of this commit? 

Thanks!!

pefernan commented:
@romartin The problem with @hasys branch is that there are some changes on master not merged on this branch I'll add them.

@romartin @jsoltes Looking into #1 & #2 right now.

I think that #3 is not an issue, since we are allowing form generation while the process isn't saved I think that it would be common having a bad formed process. Old jBPM Designer allows that too (just checked)

I can see #4 also happening (sometimes) on the morphing icons. It seems that it happens the first time you open stunner, after that it seems that the icons appear correctly.

#5 should give a response when there's any problem happening. I don't know what to say about clicking on form generation without any task selected.

jsoltes commented:
I am writting update after a call with Pere.
1. and 2. seems to be fixed after Kirill's pull-request. We agreed 3. is gonna be some future enhancement.
4. is up to you, @romartin  and @pefernan is going to fix 5.

pefernan commented:
@jsoltes @romartin added the fix for issue **5** (added notification when there are no selected items)

pefernan commented:
Jenkins execute full downstream build

romartin commented:
Hey @jsoltes @pefernan @hasys 
Added commit from @hasys  and fixed some literal. Also @pefernan added the fixes, so the only missing point is the issues with icons, but it's not related to this, as it happens as well with already existing icons, so I understand this happens on actual master when using some concrete browser versions, is it? I would say we can just a comment in [JBPM-6858](https://issues.jboss.org/browse/JBPM-6858), it seems to describe the same issue on IE, right?
Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
ok thanks guys, just rebased with master and squashed, once build is successful will do some smoke tests again and merge it.

romartin commented:
oh @pefernan look at the number of changed files... magic number! :P

romartin commented:
Jenkins execute full downstream build

pefernan commented:
@romartin yeah!
